_re2.cc

```diff
@@ -58,8 +58,8 @@ typedef struct _MatchObject2 {
   PyObject* string;
   // There are several possible approaches to storing the matched groups:
   // 1. Fully materialize the groups tuple at match time.
-  // 2. Cache allocate PyString objects when groups are requested.
-  // 3. Always allocate new PyStrings on demand.
+  // 2. Cache allocate PyBytes objects when groups are requested.
```

Comment:
While you're editing this line, maybe fix the typo "allocate"?  (Should be "allocated".)